dist_tables/querying.rst

```diff
@@ -120,3 +120,126 @@ Citus employs a two stage optimizer when planning SQL queries. The first phase i
+
+Running on all Workers
+----------------------
+
+The least granular level of execution is broadcasting a statement for execution on all workers. This is useful for viewing a properties of entire worker databases or creating UDFs uniformly throughout the cluster. For example:
+
+.. code-block:: postgresql
+
+  -- Make a UDF available on all workers
+  SELECT run_command_on_workers($cmd$ CREATE FUNCTION ..... $cmd$);
+
+  -- List the size of each worker database
+  SELECT *
+    FROM run_command_on_workers($cmd$
+      SELECT pg_size_pretty(pg_database_size('citus'))
+    $cmd$);
+
+.. note::
+
+  The :code:`run_command_on_workers` function and other manual propagation commands in this section can run only queries which return a single column and single row.
+
+Running on all Shards
+---------------------
+
+Each distributed table :code:`foo` has shards across the workers which are each tables with a naming convention of :code:`foo_n` where :code:`n` is a number. In order to run a query against a shard, the query must be run on its containing worker, and the shard name must be interpolated into the query string as an identifier (:code:`%I`).
+
+.. code-block:: postgresql
+
+  -- Get the size of table shards to determine how balanced the cluster
+  -- is. The third argument controls whether the queries are run in
+  -- parallel and defaults to true
+  SELECT *
+    FROM run_command_on_shards(
+      'my_distributed_table',
+      $cmd$ SELECT pg_size_pretty(pg_total_relation_size('%I')) $cmd$,
+      true
+    );
+
+  -- Get the total size of indices from all shards
+  SELECT pg_size_pretty(sum(result::int))
+    FROM run_command_on_shards(
+      'my_distributed_table',
+      $cmd$ SELECT pg_indexes_size('%I') $cmd$
+    );
+
+Running on all Placements
+-------------------------
+
+When using Citus- rather than streaming-replication for :ref:`dealing_with_node_failures`, each shard has replicas stored as tables on other workers. Shards and their replicas are all called *placements*. (PostgreSQL streaming replication, on the other hand, works at the database level rather than the shard level, so the notion of shard replica placements is not applicable there.)
+
+Queries that update rows ought to be run on all placements rather than simply all shards. Ordinarily update queries go through the coordinator node which ensures they apply across placements. The functions in this section bypass the normal coordinator logic, so skipping any replicas in a manual update will leave the cluster inconsistent. Conversely, read-only queries should be run on shards rather than placements or they will return duplicate results.
+
+The following are equivalent:
+
+.. code-block:: postgres
```

Comment:
I think they are synonyms, but agree that I should keep it consistent.

For reference these are the available [Pygments lexers](http://pygments.org/docs/lexers/) that enable highlighting. [This doc](https://build-me-the-docs-please.readthedocs.io/en/latest/Using_Sphinx/ShowingCodeExamplesInSphinx.html#pygments-lexers) confirms that Pygments is the underlying engine.